SmartDeviceLink/SDLEncryptionLifecycleManager.m

```diff
@@ -0,0 +1,274 @@
+//
+//  SDLEncryptionLifecycleManager.m
+//  SmartDeviceLink
+//
+//  Created by standa1 on 6/27/19.
+//  Copyright © 2019 smartdevicelink. All rights reserved.
+//
+
+#import "SDLEncryptionLifecycleManager.h"
+#import "SDLEncryptionManagerConstants.h"
+#import "SDLEncryptionConfiguration.h"
+#import "SDLServiceEncryptionDelegate.h"
+#import "SDLAsynchronousRPCRequestOperation.h"
+#import "SDLLogMacros.h"
+#import "SDLStateMachine.h"
+#import "SDLProtocolMessage.h"
+#import "SDLRPCNotificationNotification.h"
+#import "SDLOnHMIStatus.h"
+#import "SDLOnPermissionsChange.h"
+#import "SDLPermissionItem.h"
+#import "SDLPermissionConstants.h"
+#import "SDLProtocol.h"
+#import "SDLError.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLEncryptionLifecycleManager() <SDLProtocolListener>
+
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (weak, nonatomic) SDLProtocol *protocol;
+
+@property (strong, nonatomic, readwrite) SDLStateMachine *encryptionStateMachine;
+@property (copy, nonatomic, nullable) SDLHMILevel currentHMILevel;
+@property (strong, nonatomic, nullable) NSMutableDictionary<SDLPermissionRPCName, SDLPermissionItem *> *permissions;
+@property (assign, nonatomic, nullable) NSNumber *requiresEncryption;
+@property (weak, nonatomic, nullable) id<SDLServiceEncryptionDelegate> delegate;
+
+@end
+
+@implementation SDLEncryptionLifecycleManager
```

Comment:
It has come to my attention from the Android team that all of our screen managers (i.e. the `SDLTextAndGraphicManager`, `SDLSoftButtonManager`, `SDLMenuManager`, etc.) will fail to work if the policy table requires that any of the RPCs used to start those managers be encrypted. 

I think the most efficient solution to this problem is to pull out the RPCs that require encryption into an array on this manager until the encryption manager has started and then add them back to the sent queue. Of course it does mean that stale RPCs will be sent but there does not seem to be an easier solution.

Reply:
Please disregard the above comment. Dealing with the sub-managers will be dealt with in a new proposal for the next release since there will be issues with head units that do not return an `OnPermissionChange` notification.